tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,199 @@
+import copy
+import os
+import pytest
+import time
+
+from configuration import available_ports
+from common import ProviderOptions, Protocols, data_bytes, Ciphers, Certificates
+from fixtures import managed_process
+from providers import S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name
+
+# Test Vectors from https://tools.ietf.org/html/rfc8448#section-4 
+shared_psk_identity = '2c035d829359ee5ff7af4ec900000000262a6494dc486d2c8a34cb33fa90bf1b00'\
+                      '70ad3c498883c9367c09a2be785abc55cd226097a3a982117283f82a03a143efd3'\
+                      'ff5dd36d64e861be7fd61d2827db279cce145077d454a3664d4e6da4d29ee03725'\
+                      'a6a4dafcd0fc67d2aea70529513e3da2677fa5906c5b3f7d8f92f228bda40dda72'\
+                      '1470f9fbf297b5aea617646fac5c03272e970727c621a79141ef5f7de6505e5bfb'\
+                      'c388e93343694093934ae4d357'
+
+shared_psk_secret = '4ecd0eb6ec3b4d87f5d6028f922ca4c5851a277fd41311c9e62d2c9492e1c4f3'
+
+s2n_known_value_psk = '2c035d829359ee5ff7af4ec900000000262a6494dc486d2c8a34cb33fa90bf1b00'\
+                      '70ad3c498883c9367c09a2be785abc55cd226097a3a982117283f82a03a143efd3'\
+                      'ff5dd36d64e861be7fd61d2827db279cce145077d454a3664d4e6da4d29ee03725'\
+                      'a6a4dafcd0fc67d2aea70529513e3da2677fa5906c5b3f7d8f92f228bda40dda72'\
+                      '1470f9fbf297b5aea617646fac5c03272e970727c621a79141ef5f7de6505e5bfb'\
+                      'c388e93343694093934ae4d357,'\
+                      '4ecd0eb6ec3b4d87f5d6028f922ca4c5851a277fd41311c9e62d2c9492e1c4f3,'\
+                      'SHA256'
+
+s2n_known_value_psk_2 = '2c035d829359ee5ff7af4ec900000000262a6494dc486d2c8a34cb33fa90bf1b00'\
+                        '70ad3c498883c9367c09a2be785abc55cd226097a3a982117283f82a03a143efd3'\
+                        'ff5dd36d64e861be7fd61d2827db279cce145077d454a3664d4e6da4d29ee03725'\
+                        'a6a4dafcd0fc67d2aea70529513e3da2677fa5906c5b3f7d8f92f228bda40dda72'\
+                        '1470f9fbf297b5aea617646fac5c03272e970727c621a79141ef5f7de6505e5bfb'\
+                        'c388e93343694093934ae4d357,'\
+                        '4ecd0eb6ec3b4d87f5d6028f922ca4c5851a277fd41311c9e62d2c9492e1c4f3,'\
+                        'SHA384'
+
+# Arbitrary Test vectors
+s2n_psk_parameters = 's2n_psk_test_identity,'\
+                      'a6dadae4567876,'\
+                      'SHA384'\
+
+s2n_psk_parameters_2 = 's2n_psk_test_identity_2,'\
+                      'a64dafcd0fc67d2a,'\
+                      'SHA256'\
+
+openssl_psk_parameters = ['-psk_identity', shared_psk_identity, '--psk', shared_psk_secret]
+
+S2N_TEST_SUITE = [
```

Comment:
To help with readability, consider [named tuples](https://stackoverflow.com/questions/2970608/what-are-named-tuples-in-python), or even a class or dictionary.